notebooks/tvb_flattenramp/TVB_flattenramp_notebook.ipynb

```diff
@@ -0,0 +1,819 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "raw_files = glob.glob('*_raw.fits')\n",
+    "\n",
+    "for fil in raw_files:\n",
+    "    !crds bestrefs --files $fil --sync-references=1 --update-bestrefs;"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4.2: Recalibrating affected data with 'ramp fitting' step turned off in `calwf3` <a id=\"4.2\"></a>\n",
+    "\n",
+    "Next, we run calwf3 on the affected file *ibohbfb9q* with CRCORR=OMIT to perform all steps except the ramp fitting."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "os.rename('ibohbfb9q_ima.fits', 'ibohbfb9q_ima_orig.fits')\n",
+    "os.rename('ibohbfb9q_flt.fits', 'ibohbfb9q_flt_orig.fits')\n",
+    "\n",
+    "raw_file = 'ibohbfb9q_raw.fits'\n",
+    "raw=fits.open(raw_file,mode='update')\n",
+    "raw[0].header['CRCORR'] ='OMIT'\n",
+    "raw.flush()\n",
+    "\n",
+    "wfc3tools.calwf3(raw_file );\n",
+    "\n",
+    "os.rename('ibohbfb9q_flt.fits', 'ibohbfb9q_flt_norampfit.fits')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4.3: Remove Median Background per read from resulting IMA <a id=\"4.3\"></a>\n",
+    "Next, the median background level from each read of the resulting IMA is subtracted from the read, and is later added back to the full exposure to preserve pixel statistics.\n",
+    "\n",
+    "You can define a specific subregion for stats, as well as add additional lower or upper thresholds to fine-tune the calculation of the sky median value. Here, we use the median value of the entire image minus the 5 pixel wide overscan regions with no sigma clipping.\n",
+    "\n",
+    "This technique works best for relatively sparse fields, where the median value of the ramp is equivalent to the sky level."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ima = fits.open(raw_file.replace('raw','ima'), mode ='update')\n",
+    "\n",
+    "### You can define a subregion for stats. Here we use the whole image, minus the 5 pixel wide overscan regions\n",
+    "stats_region =[[5,1014], [5,1014]]\n",
+    "slx = slice(stats_region[0][0], stats_region[0][1])\n",
+    "sly = slice(stats_region[1][0], stats_region[1][1])\n",
+    "\n",
+    "### Subtract the median countrate from each read and add back the full exposure countrate to preserve pixel statistics\n",
+    "   \n",
+    "total_countrate = np.median(ima['SCI',1].data[sly, slx])\n",
+    "    \n",
+    "for i in range(ima[0].header['NSAMP'] -1):\n",
+    "    med = np.median(ima['SCI',i+1].data[sly, slx])\n",
+    "    ima['SCI',i+1].data += total_countrate - med\n",
+    "    print('%s, [SCI,%d], median_bkg: %.2f' %(raw_file, i+1, med))    \n",
```

Comment:
Might make this a table or plot (sci ext vs median bkg), but it if doesn't look better, don't worry about it